impl/maven-cli/src/main/java/org/apache/maven/cling/invoker/LookupInvoker.java

```diff
@@ -308,13 +309,13 @@ protected void createTerminal(C context) {
         if (context.terminal == null) {
             MessageUtils.systemInstall(
                     builder -> {
-                        builder.streams(
-                                context.invokerRequest.in().orElse(null),
-                                context.invokerRequest.out().orElse(null));
-                        builder.systemOutput(TerminalBuilder.SystemOutput.ForcedSysOut);
-                        // The exec builder suffers from https://github.com/jline/jline3/issues/1098
-                        // We could re-enable it when fixed to provide support for non-standard architectures,
-                        // for which JLine does not provide any native library.
+                        if (context.invokerRequest.embedded()) {
+                            builder.streams(
+                                    context.invokerRequest.stdIn().orElse(InputStream.nullInputStream()),
+                                    context.invokerRequest.stdOut().orElse(System.out));
```

Comment:
Why not `OutputStream.nullOuputStream()` ? If the user does not want output, let's not print it out.
Else, it will be difficult to work around.  And user can simply pass in `System.out` if he wants.